### PR TITLE
fix: allow fine-granular resource CPU settings

### PR DESCRIPTION
**Pull request by adwk67**
# Description

fixes https://github.com/stackabletech/spark-k8s-operator/issues/363.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Comment by adwk67** [comment]
Jenkins: :runner: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/spark-k8s-operator-it-custom/133/

**Review by sbernauer — APPROVED** [review]
This is exactly what I was thinking of, awesome!
Change LGTM, I'm currently thinking if we can improve any tests around this :thinking: At least we already have the resources kuttl test :)

**Comment by sbernauer** [comment]
We *might* want to mark it as breaking in the changelog:

BREAKING: Specified CPU resources are now applied correctly (instead of rounding it to the next whole number). This might affect your jobs, as they now e.g. only have 200m CPU resources available instead of the 1000m it had so far and they might slow down significantly

**Review by sbernauer — APPROVED** [review]

**Comment by adwk67** [comment]
Yes, good point. Have added note to changelog.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,7 +8,14 @@ All notable changes to this project will be documented in this file.
 
 - Update Rust dependency versions, most notably operator-rs 0.67.1 ([#401])
 
+### Fixed
+
+- BREAKING (behaviour): Specified CPU resources are now applied correctly (instead of rounding it to the next whole number).
+  This might affect your jobs, as they now e.g. only have 200m CPU resources requested instead of the 1000m it had so far,
+  meaning they might slow down significantly([#408]).
```
Comment (sbernauer) [inline CHANGELOG.md:15]:
```suggestion
  meaning they might slow down significantly ([#408]).
```

**Review by sbernauer — APPROVED** [review]

**Review by sbernauer — APPROVED** [review]

**Pull request merged**